### DISCUSSION TITLE
wav file not playing outside of game environment

### DISCUSSION THREAD

**Discussion by rohrlich** [hajimehoshi/ebiten#1784] [Q&A]
I want to play a wav file outside of the game environment. There are no errors but I don't here the sound unless I call ebiten.RunGame(). So what needs to be setup to get this to work. Is the key in the UIContext code?

**Comment by hajimehoshi** [comment]
This is intended and there is no way to play audio before `RunGame` so far. If you just want to play an audio, use github.com/hajimehoshi/oto directly, but in this case you cannot use Ebiten.

**Comment by rohrlich** [comment]
Thank you for the quick reply!

**Comment by rohrlich** [comment]
Though I can run the oto code, I haven't found a way to get a wav file stream to pass to NewPlayer. Is there a decode method that will return a Reader I can pass to NewPlayer or other solution?

**Reply by hajimehoshi** [reply]
Have you tried https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2/audio/wav#DecodeWithSampleRate?

**Reply by rohrlich** [reply]
Yes that is exactly what I tried, should have mentioned that. I get linker errors, presumably because the audio/wav import is including the oto which I am also importing. Here are the first couple of errors:
# command-line-arguments
/usr/local/go/pkg/tool/darwin_amd64/link: running clang failed: exit status 1
duplicate symbol '_oto_render' in:
    /var/folders/59/7ys8nmzs4dz8qppbdjpn8xfc0000gn/T/go-link-834648450/000000.o
    /var/folders/59/7ys8nmzs4dz8qppbdjpn8xfc0000gn/T/go-link-834648450/000013.o
duplicate symbol '_oto_setGlobalPause' in:
    /var/folders/59/7ys8nmzs4dz8qppbdjpn8xfc0000gn/T/go-link-834648450/000000.o
    /var/folders/59/7ys8nmzs4dz8qppbdjpn8xfc0000gn/T/go-link-834648450/000013.o

**Reply by hajimehoshi** [reply]
Which version of Ebiten are you using?

**Reply by rohrlich** [reply]
V2

**Reply by hajimehoshi** [reply]
v2.1.4, or the latest commit?

The error message seems to mix oto v1 and oto v2.

**Comment by rohrlich** [comment]
From my go.mod:
	github.com/hajimehoshi/ebiten/v2 v2.1.4
	github.com/hajimehoshi/oto/v2 v2.0.0-alpha.2

**Reply by hajimehoshi** [reply]
OK so ebiten v2.1.4 uses oto v1, while you use oto v2. Please try to use oto v1.

**Reply by rohrlich** [reply]
ok, I'll try that. Thanks!

**Reply by rohrlich** [reply]
Working, thank you so much!
